app/services/katello/rhsm_fact_parser.rb

```diff
@@ -0,0 +1,151 @@
+module Katello
+  REDHAT_ATOMIC_HOST_DISTRO_NAME = "Red Hat Enterprise Linux Atomic Host".freeze
+  REDHAT_ATOMIC_HOST_OS = "RedHat_Enterprise_Linux_Atomic_Host".freeze
+
+  class RhsmFactParser < ::FactParser
+    def architecture
+      name = facts['lscpu.architecture'] || facts['uname.machine']
+      name = "x86_64" if name == "amd64"
+      name = "i386" if name == "i686"
+      Architecture.where(:name => name).first_or_create if name.present?
+    end
+
+    def model
+      if facts['virt::is_guest'] == "true"
+        name = facts['lscpu.hypervisor_vendor']
+      else
+        name = facts['dmi.system.product_name']
+      end
+      ::Model.where(:name => name.strip).first_or_create if name.present?
+    end
+
+    def support_interfaces_parsing?
+      true
+    end
+
+    def get_facts_for_interface(interface)
+      {
+        'link' => true,
+        'macaddress' => get_rhsm_mac(interface),
+        'ipaddress' => get_rhsm_ip(interface),
+      }
+    end
```

Comment:
Making a note so we don't forget: this is IPv4 only but it should really also parse the IPv6 fact. I'm not sure if you keep any notes for the refactor, but @ShimShtein ran into this today.

Reply:
Thanks for the Info. I will update commit accordingly.

Reply:
FYI: in Katello it's also IPv4-only so it'd be fine (perhaps even better) to do it as a separate issue.